README.md

```diff
@@ -263,9 +263,9 @@ additional information._
 
 ### 5.2 Delegate Design
 
-The `DelegateInterface` defines a single method that accepts a request and
-returns a response. The delegate interface must be implemented by any middleware
-dispatcher that uses middleware implementing `MiddlewareInterface`.
+A delegates is a representative of the middleware dispatcher, whose only purpose
+is to encapsulate the remaining request processing. The `DelegateInterface` defines a
+`process` method therefor, which accepts a request and returns a response.
```

Comment:
> but the language is totally different

I rarely see problems in changing writing style.

---

> This is done by dispatching the next available middleware, or if none is available, by creating an empty response.

Or throw an exception, or … – I believe such an list will always be incomplete.

Perhaps, we may just add examples, similar to the middleware section:

> _as already suggested:_
> A delegates is a representative of the middleware dispatcher, whose only purpose
> is to encapsulate the remaining request processing. The `DelegateInterface` defines a
> `process` method therefor, which accepts a request and returns a response.
> 
> _now the new part:_
> The delegate may:
> - Dispatching the next available middleware.
> - Dispatching a specific middleware, depending on the provided request.
> - Creating an empty response.
> - Creating an response, depending on the provided request.
> - …

For example:
* The second list item also covers router middlewares.
* The forth item also covers preconfigured XML/JSON responses, based on the `Accept` header of the request.

---

Btw, I would like to preserve the concrete `process` name method, it makes the following discussions easier to grasp. For example the _Why does the delegate conflict with middleware?_ section – If the reader already knows that the method name is also `process` we gain readability in my opinion.

---

While the language might be different, I like:
> whose (only) purpose is to _encapsulate the remaining request processing_

The alternative I can imagine, would be something like: 
> whose (only) purpose is to _delegate requests (back?) to the middleware dispatcher_ :disappointed: